hyperdome/server/web.py

```diff
@@ -124,106 +121,94 @@ def shutdown(slug_candidate):
-        def poll_connected_guest():
-            counselor_id = request.form["counselor_id"]
-            guest_key = self.guest_keys.pop(counselor_id, "")
-            return guest_key
-
-        @app.route("/counseling_complete", methods=["POST"])
-        def counseling_complete():
-            sid = request.form["user_id"]
-            if sid not in self.active_chat_user_map:
-                return "no active chat", 404
-            other_user = self.active_chat_user_map[sid]
-            self.active_chat_user_map.pop(sid)
-            self.pending_messages.pop(sid, "")
+                return jsonify(error="no counselors available")
+                # TODO: implement guest queue
+            counselor_id = secrets.choice(counselors)
+            self.counselors[counselor_id]["connected_guest"] = guest_id
+            return jsonify(counselor=self.counselors[counselor_id])
+
+        @app.route("/hyperdome/api/v1/guests/<string:counselor_id>", methods=["GET"])
+        def poll_connected_guest(counselor_id):
             try:
-                self.active_chat_user_map[other_user] = ""
+                guest_id = self.counselors[counselor_id]["connected_guest"]
+                guest = self.guests[guest_id]
             except KeyError:
-                pass
-            if sid in self.counselors_available:
-                counselor_id = sid
-            elif other_user in self.counselors_available:
-                counselor_id = other_user
+                abort(404, message="Connected guest not found")
             else:
-                return "Counselor has left the chat"
-            self.counselors_available[counselor_id] += 1
-            return "Chat Ended"
-
-        @app.route("/counselor_signout", methods=["POST"])
-        def counselor_signout():
-            sid = request.form["user_id"]
-            self.counselors_available.pop(sid, "")
-            self.counselor_keys.pop(sid, "")
-            return "Success"
-
-        @app.route("/counselor_signin", methods=["POST"])
-        def counselor_signin():
-            username = request.form["username"]
-            session_counselor_key = request.form["pub_key"]
-            signature = request.form["signature"]
+                return jsonify(guest=guest)
+
+        @app.route("/hyperdome/api/v1/counselors/", methods=["POST"])
+        def counselor_login():
+            username = request.json["username"]
+            session_counselor_key = request.json["pub_key"]
+            signature = request.json["signature"]
             signature = base64.urlsafe_b64decode(signature)
             counselor = models.Counselor.query.filter_by(name=username).first_or_404()
             if not counselor.verify(signature, session_counselor_key):
-                self.__log.info(
+                self.__log.warning(
                     f"attempted counselor login failed verification {username=}"
                 )
-                return "Bad signature", 401
+                request
```

Comment:
:?
What's this line doing?

Reply:
If I were to guess, that was me probing the properties of the request variable to see if there was an easier way to do something, I'm surprised none of the 3 linters I'm running isn't complaining about that